### ISSUE TITLE
`migrations.removeUnused` seems not to be working

### ISSUE THREAD

**Issue by timche** [bfred-it/webext-options-sync#10]
Thanks for this great library first of all!

I just tried out `migrations.removeUnused`, but it seems like it doesn't remove the unused properties.

Initial options definition:

```js
new OptionsSync().define({
  defaults: {
    foo: true
  },
  migrations: [OptionsSync.migrations.removeUnused]
});
```

![screen shot 2018-03-19 at 15 17 11](https://user-images.githubusercontent.com/8075074/37600786-9b51b7ea-2b88-11e8-9367-0c5fbdcb5e3e.png)

Removing `foo` from the `defaults`, running the migration and refreshing `background.js`:

```js
new OptionsSync().define({
  defaults: {},
  migrations: [OptionsSync.migrations.removeUnused]
});
```

![screen shot 2018-03-19 at 15 17 58](https://user-images.githubusercontent.com/8075074/37600833-bbd746e2-2b88-11e8-92a2-5bd5dc1a8315.png)

As you can see, `foo` is still there.

I'm not really sure if I'm just missing something?

**Comment by bfred-it** [comment]
Nice catch! I was misusing `Object.assign` This is fixed in 0.15.1.

**Comment by timche** [comment]
Thanks a lot! :)

P.S. Your library is powering [this browser extension](https://chrome.google.com/webstore/detail/faceit-enhancer/mokknliiomknodkdmpcellamkopbdmao?hl=en) I've built.
